src/lib.rs

```diff
@@ -1,35 +1,55 @@
-//! Match strings against a simple wildcard pattern.
-//! Tests a wildcard pattern `p` against an input string `s`. Returns true only when `p` matches the entirety of `s`.
+//! Match strings against a simple wildcard pattern. Tests a wildcard
+//! pattern `p` against an input string `s`. Returns true only when `p`
+//! matches the entirety of `s`.
 //!
-//! See also the example described on [wikipedia](https://en.wikipedia.org/wiki/Matching_wildcards) for matching wildcards.
+//! See also the example described on
+//! [wikipedia](https://en.wikipedia.org/wiki/Matching_wildcards) for
+//! matching wildcards.
 //!
 //! No escape characters are defined.
 //!
 //! - `?` matches exactly one occurrence of any character.
-//! - `*` matches arbitrary many (including zero) occurrences of any character.
+//! - `*` matches arbitrary many (including zero) occurrences of any
+//!   character.
 //!
 //! Examples matching wildcards:
 //! ``` rust
-//! # extern crate wildmatch; use wildmatch::WildMatch;
-//! assert!(WildMatch::new("cat").matches("cat"));
-//! assert!(WildMatch::new("*cat*").matches("dog_cat_dog"));
-//! assert!(WildMatch::new("c?t").matches("cat"));
-//! assert!(WildMatch::new("c?t").matches("cot"));
+//! # extern crate wildmatch; use wildmatch::GlobWildMatch;
+//! assert!(GlobWildMatch::new("cat").matches("cat"));
+//! assert!(GlobWildMatch::new("*cat*").matches("dog_cat_dog"));
+//! assert!(GlobWildMatch::new("c?t").matches("cat"));
+//! assert!(GlobWildMatch::new("c?t").matches("cot"));
 //! ```
 //! Examples not matching wildcards:
 //! ``` rust
+//! # extern crate wildmatch; use wildmatch::GlobWildMatch;
+//! assert!(!GlobWildMatch::new("dog").matches("cat"));
+//! assert!(!GlobWildMatch::new("*d").matches("cat"));
+//! assert!(!GlobWildMatch::new("????").matches("cat"));
+//! assert!(!GlobWildMatch::new("?").matches("cat"));
+//! ```
+//!
+//! You can specify custom `char` values for the single and multi-character
+//! wildcards. For example, to use `%` as the multi-character wildcard and
+//! `_` as the single-character wildcard:
+//! ```rust
 //! # extern crate wildmatch; use wildmatch::WildMatch;
-//! assert!(!WildMatch::new("dog").matches("cat"));
-//! assert!(!WildMatch::new("*d").matches("cat"));
-//! assert!(!WildMatch::new("????").matches("cat"));
-//! assert!(!WildMatch::new("?").matches("cat"));
+//! assert!(WildMatch::<'%', '_'>::new("%cat%").matches("dog_cat_dog"));
 //! ```
 
 use std::fmt;
 
+/// A wildcard matcher using `*` as the multi-character wildcard and `?` as
+/// the single-character wildcard.
+pub type GlobWildMatch = WildMatch<'*', '?'>;
```

Comment:
This would be a breaking change. Wouldn't it alse be possible to keep the type `WildMatch` for the default one and rename the concrete implementation to something like `WildMatchPattern`? This would not break every project out there using wildmatch?